includes/sanitizers/class-amp-audio-sanitizer.php

```diff
@@ -135,10 +163,24 @@ public function sanitize() {
 			if ( empty( $sources ) ) {
 				$this->remove_invalid_child( $node );
 			} else {
-				$noscript = $this->dom->createElement( 'noscript' );
-				$new_node->appendChild( $noscript );
 				$node->parentNode->replaceChild( $new_node, $node );
-				$noscript->appendChild( $old_node );
+
+				if ( ! empty( $this->args['add_noscript_fallback'] ) ) {
+					$noscript = $this->dom->createElement( 'noscript' );
+					$noscript->appendChild( $old_node );
+					$new_node->appendChild( $noscript );
+
+					// Remove all non-allowed attributes preemptively to prevent doubled validation errors.
+					$disallowed_attributes = array();
+					foreach ( $old_node->attributes as $attribute ) {
+						if ( ! isset( $this->noscript_fallback_allowed_attributes[ $attribute->nodeName ] ) ) {
+							$disallowed_attributes[] = $attribute->nodeName;
+						}
+					}
+					foreach ( $disallowed_attributes as $disallowed_attribute ) {
+						$old_node->removeAttribute( $disallowed_attribute );
+					}
```

Comment:
Why not iterate only once and immediately remove the attributes instead of populating first and then doing another round of removing? If the problem is remove attributes modifying the `$old_node->attributes` property, then we could assign it to another variable to use.

Reply:
The problem is that the attributes are a live DOM list and if a node is removed while iterating then a node gets skipped. This pattern of gathering up the list of nodes and then removing them on a second iteration can be seen elsewhere in the codebase. It's not ideal.